### PR TITLE
Fixes pomodoro.

### LINKED ISSUE #245: [PE-D] Timer gets stuck
Time does not run at all when the time added is too large.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: ShengXue97/ped#7

### PR DESCRIPTION
**Pull request by Hardoh**
Stats is successfully updated between exits and reopens.

Big values for tm/ is now limited to 24hours.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/PomCommand.java [hunk 1]
```diff
@@ -100,13 +116,18 @@ public CommandResult execute(Model model) throws CommandException {
...
 
+        String resultText =
+                model.getPomodoroTask() == null
+                        ? "Pomming task: " + taskToPom.toString()
+                        : "Switched pom task: " + taskToPom.toString();
+
         // Update the pomodoro model
         model.setPomodoroTask(taskToPom);
 
         System.out.println("Hardy: " + (pm.getDefaultStartTime()));
```
Comment (BransonNg) [inline src/main/java/seedu/address/logic/commands/PomCommand.java:127]: Should use logger for useful information. they actually recommend logging some stuff

File: src/main/java/seedu/address/MainApp.java [hunk 2]
```diff
@@ -291,6 +291,7 @@ public void stop() {
                 "============================ [ Stopping Address Book ] =============================");
         try {
             storage.savePomodoro(model.getPomodoro());
+            storage.saveStatistics(model.getStatistics());
```
Comment (BransonNg) [inline src/main/java/seedu/address/MainApp.java:294]: mm, how do u ensure that the stats are updated when saving?

File: src/main/java/seedu/address/logic/PomodoroManager.java [hunk 3]
```diff
@@ -508,5 +512,21 @@ public void handleExit() {
         }
         float timeInMinutes = timeSeconds.floatValue() / 60;
         model.setPomodoroTimeLeft(timeInMinutes);
+
+        // Update statistics so far
+        // Update stats
+        updateStatistics(model);
```
Comment (BransonNg) [inline src/main/java/seedu/address/logic/PomodoroManager.java:518]: is this the part that handles the exiting?

File: src/main/java/seedu/address/logic/commands/PomCommand.java [hunk 4]
```diff
@@ -71,18 +75,30 @@ public CommandResult execute(Model model) throws CommandException {
         PomodoroManager pm = model.getPomodoroManager();
 
         if (pomType == POM_TYPE.PAUSE) {
-            pm.pause();
-            return new PomCommandResult(PAUSE_MESSAGE, null, 0, model, -1, null, pomType);
+            try {
+                pm.pause();
+                return new PomCommandResult(PAUSE_MESSAGE, null, 0, model, -1, null, pomType);
+            } catch (NullPointerException ne) {
+                return new PomCommandResult(NO_POM, null, 0, model, -1, null, pomType);
```
Comment (BransonNg) [inline src/main/java/seedu/address/logic/commands/PomCommand.java:82]: also try to reduce the magic fields

**Review by BransonNg — APPROVED** [review]
lgtm!

File: src/main/java/seedu/address/logic/PomodoroManager.java (hunk 3 above)
Comment (Hardoh) [reply src/main/java/seedu/address/logic/PomodoroManager.java:518]: Yes it is!

**Pull request merged**